clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -904,16 +904,22 @@ private void sendProduceRequest(long now, int destination, short acks, int timeo
         }
 
         String transactionalId = null;
+        boolean canUseTransactionV2AboveVersion = true;
```

Comment:
nit: can we rename this to isTransactionV2Enabled to keep it consistent and avoid confusion

Reply:
Ahh that makes sense, just an idea but should we name it canUseLatestStableVersion then? And add comments about when to use which?

Reply:
It seems like this boolean is also flagging that you can just use the latest produce version without extra logic?

Reply:
> we name it canUseLatestStableVersion

It is not clear what is the expected behavior if false, which is the key reason to have the flag.